diffcp/cone_program.py

```diff
@@ -165,12 +181,30 @@ def derivative(dA, db, dc, **kwargs):
             [-dA, None, np.expand_dims(db, -1)],
             [-np.expand_dims(dc, -1).T, -np.expand_dims(db, -1).T, None]
         ])
-        # can ignore w since w = 1
         rhs = dQ @ pi_z
         if np.allclose(rhs, 0):
             dz = np.zeros(rhs.size)
-        else:
+        elif mode == "dense":
+            dz = _diffcp._solve_derivative_dense(M, MT, rhs)
+        elif mode == "sparse":
+            rho = kwargs.get("rho", 1e-6)
+            it_ref_iters = kwargs.get("it_ref_iters", 10)
+            M_iref = sparse.bmat([
+                [-sparse.eye(N), M],
+                [MT, rho * sparse.eye(N)]
+            ])
+            solve = splinalg.factorized(M_iref.tocsc())
+            rhs = MT @ rhs
+            dz = np.zeros(N)
+            # iterative refinement
+            for _ in range(it_ref_iters):
+                residual = rhs - MT @ (M @ dz)
+                if np.linalg.norm(residual) <= 1e-10:
+                    break
+                dz = dz + solve(np.append(np.zeros(N), residual))[N:]
+        elif mode == "lsqr":
```

Comment:
nit: `else`, since this is the only other option

Reply:
done